client/services/TranscriptService.js

```diff
@@ -18,15 +18,18 @@
  */
 
 tie.factory('TranscriptService', [
-  'EventObjectFactory', function(EventObjectFactory) {
-    var transcript = [];
+  'EventObjectFactory', 'TranscriptObjectFactory', function(EventObjectFactory, TranscriptObjectFactory) {
+    var transcript = TranscriptObjectFactory.create();
 
     return {
+      recordSnapshot: function(snapshot) {
+        transcript.recordSnapshot(snapshot);
+      },
       recordSolution: function(code) {
-        transcript.push(EventObjectFactory.createCodeSubmittedEvent(code));
+        transcript.recordEvent(EventObjectFactory.createCodeSubmittedEvent(code));
```

Comment:
Really, I think it's fine to drop these old functions. We can add them later if needed but let's not have dead code, especially now that you have an actual use for the transcript.